modules/nw-sriov-troubleshooting.adoc

```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * networking/hardware_networks/configuring-sriov-device.adoc
+
+[id="nw-sriov-troubleshooting_{context}"]
+= Troubleshooting SR-IOV configuration
+
+After following the procedure to configure an SR-IOV network device, the following sections address some error conditions.
+
+To display the state of nodes, run the following command:
+
+[source,terminal]
+----
+$ oc get sriovnetworknodestates -n openshift-sriov-network-operator <node_name>
+----
+
+where `<node_name>` specifies the name of a node with an SR-IOV network device.
+
+.Error output: Cannot allocate memory
+[source,terminal]
+----
+"lastSyncError": "write /sys/bus/pci/devices/0000:3b:00.1/sriov_numvfs: cannot allocate memory"
+----
+
+When a node indicates that it cannot allocate memory, check the following items:
+
+* Confirm that the BIOS for the node is configured to enable SR-IOV.
+
+* Confirm that the BIOS for the node is configured to enable VT-d.
```

Comment:
Add another item:

Sometimes, the BIOS is not providing enough MMIO space for VFs. Contact your hardware vendor for firmware or bios update. As a workaround solution, one can pass "pci=realloc" to kernel argument [link-to-mco-doc-on-updating-kernel-argument](https://docs.openshift.com/container-platform/4.7/nodes/nodes/nodes-nodes-working.html#nodes-nodes-kernel-arguments_nodes-nodes-working)

Reply:
As far as I understand when `pci=realloc` is enabled, there are no restrictions to memory allocation.
I think this is why it makes it works in the first place.
I am raising this because it sounds like something that may degrade the stability of the node,
What you think?

Reply:
@ormergi, that question is to @zshi-redhat, correct?  I left the `pci=realloc` suggestion out because it seemed like the proper fix was to reconfigure the BIOS.

@zshi-redhat, if we keep the `pci=realloc` suggestion, does the customer receive the same error message (cannot allocate memory) or is there some other message--and is the message visible after `oc get sriovnetworknodestates...` command or some other way?

Reply:
@ormergi I thought your original request is to add the `pci=realloc` workaround in the doc, thus the comment above.
I didn't know the potential instability when enabling the `pci=realloc`, if that's the case, it sounds to me that we should not provide such workaround which may cause any other instability issues. so let's not add it in the doc.

@mikemckiernan  my understanding is the error message is the same `cannot allocate memory`, it can be observed from `oc get sriovnetworknodestate <node-name>`.

Reply:
I pushed the PR again to address Or's other comment.  My instinct, at this point, is leave `pci=realloc` for another update.  At that time, maybe we'll have information about when and why to try that fix instead of others.  That said, I'm still open to suggestion if we have that information now.